articles/introduction-to-theory-of-computation/index.md

```diff
@@ -0,0 +1,132 @@
+Real world computers perform computations that by nature run like mathematical models to solve problems in systematic ways. 
+The essence of theory of computation is to help develop mathematical and logical models that run efficiently and to a point of halting.Since all machines that implement logic apply TOC, studying TOC gives learners an insight of computer hardware and software limitations
+<!--more-->
+
+### Key considerations of computational problems
+1. What can and cannot be computed.
+2. Speed of such computations.
+3. The amount of memory in use during such computations.
+
+#### Importance of Theory of computation
+The theory of computation forms the basis for:
+* Writing efficient algorithms that run in computing devices
+* Programming language research and their development.
+* Efficient Compiler design and construction
+<!--more-->
+The Theory of Computation is made up of 3 branches i.e
+- Automata Theory
+- Computability Theory
+- Complexity Theory
+
+
+### Basic of Automata Theory
+This branch of theoretical computer science was developed by mathematicians to simplify the logic of computation by using well defined abstract computational devices(models).
+[Automata Theory](https://en.wikipedia.org/wiki/Automata_theory) is basically the study of abstract computational devices and forms a formal framework for design and analysis new types of computing devices such as bio computers and quantum computers.
+These models are essential in several areas of computation (applied and theoretical fields). 
+<!--more-->
+Through automata, we learn how  problems and compute functions are solved by use of automatons. 
+An Automaton is a logical unit of machine that operate singly on an input and follow a defined pattern or configuration so as to produce a desired output.
+
+A basic computation performed on/by an automaton is defined by the following features:
+
++ A set of input symbols
++	Configuration States
++ Output
+
+
+#### Branches of Automata Theory 
+
+- **Finite Automata(FA)**- Is a kind of a computer though inferior in its computation ability. This model is fit for devices with limited memory. It is a simple abstract machine with 5 elements that define its functioning and processing of problems.
+<!--more-->
+Generally, a [Finite Automaton(FA)](https://www.geeksforgeeks.org/introduction-of-finite-automata/) is a finite collection of states with rules (transition functions) for traversing through the states depending on the input symbol.
+FA accepts or rejects input strings while reading the strings from left to right.
+**The tuples are –**
++ **Q:** *Finite set of states*
++ **∑:** *Set of input symbols*
++ **q:** *Initial state*
++ **F:** *Set of final states*
++ **δ:** *Transition function*
+
+<!--more-->
+Finite Automata is useful in building text editors / text preprocessors.  FA are poor models of computer in that they can only perform simple computational tasks. Nonetheless, they are essential parts of algorithms and computers.
```

Comment:
Add a period of `FA are poor models of computers`. and remove `in that`. `They can only ... `, frame it as a new sentence.